docs/reference/css_guidelines.rst

```diff
@@ -0,0 +1,151 @@
+Contributing CSS to Wagtail
```

Comment:
+1 - having that doc under 'How to' seems a bit buried too.

(Whatever we decide, I think we should put this new doc together with the old one, rather than having it separate under Reference.)